portal/src/OnboardingSurveyScreen.tsx

```diff
@@ -697,48 +714,69 @@ function Step4(_props: StepProps) {
   const prefix = "step4";
   const reasonChoiceGroup = "reasonChoiceGroup";
   const reasonChoices = ["Auth", "SSO", "Security", "Portal", "Other"];
-  const reasonChoicesFromLocalStorage = getFromLocalStorage("reasonChoices");
+  const reasonChoicesFromLocalStorage = getFromLocalStorage("use_cases") as
+    | UseCase[]
+    | undefined;
   const defaultReasonChoices: string[] = (
     reasonChoicesFromLocalStorage === undefined
       ? []
-      : reasonChoicesFromLocalStorage
+      : reasonChoicesFromLocalStorage.filter((elem) => typeof elem === "string")
   ) as string[];
   const [reasonChoicesState, setReasonChoicesState] =
     useState(defaultReasonChoices);
   const empty = useMemo(() => {
     return reasonChoicesState.length === 0;
   }, [reasonChoicesState]);
   const { renderToString } = useContext(Context);
-  const otherReasonFromLocalStorage = getFromLocalStorage("otherReason");
-  const [otherReason, setOtherReason] = useState(
-    otherReasonFromLocalStorage === undefined
-      ? ""
-      : (otherReasonFromLocalStorage as string)
-  );
+  let otherReasonFromLocalStorage = "";
+  if (reasonChoicesFromLocalStorage !== undefined)
+    reasonChoicesFromLocalStorage.forEach((elem) => {
+      if (typeof elem !== "string")
+        otherReasonFromLocalStorage = elem.other_reason;
+    });
+  const [otherReason, setOtherReason] = useState(otherReasonFromLocalStorage);
   const navigate = useNavigate();
+  const capture = useCapture();
   useEffect(() => goToFirstUnfilled(4, navigate), [navigate]);
   const onClickNext = useCallback(
     (e) => {
       e.preventDefault();
       e.stopPropagation();
-      setLocalStorage("reasonChoices", reasonChoicesState);
-      if (otherReason !== "") setLocalStorage("otherReason", otherReason);
+      const constructUseCases = reasonChoicesState as UseCase[];
+      if (constructUseCases.includes("Other")) {
+        const i = constructUseCases.indexOf("Other", 0);
+        constructUseCases[i] = { other_reason: otherReason };
+      }
+      setLocalStorage("use_cases", reasonChoicesState);
+      capture("onboardingSurvey.set-use_cases");
+      const final_posthog_survey_json_obj = JSON.parse(
+        localStorage.getItem(localStorageKey)!
+      );
+      if ("phone_number" in final_posthog_survey_json_obj)
+        final_posthog_survey_json_obj.phone_number =
+          final_posthog_survey_json_obj.phone_number.e164;
+      capture("onboardingSurvey.set-completed-survey", {
```

Comment:
I remember that we need to store the survey json into custom attributes.
Are you going to handle that in this issue? https://linear.app/authgear/issue/DEV-1422/redirect-new-users-to-onboarding-survey-after-signing-up

Reply:
my bad, missed this part, will edit the pull request